### ISSUE TITLE
Mula errors: Task has been finished, but no results found in bytes 

### ISSUE THREAD

**Issue by praseodym** [minvws/nl-kat-coordination#1385]
On kattest01, Mula repeatedly logs errors like these:
```
[2023-07-12 14:54:32 +0000] [ERROR] [scheduler.schedulers.scheduler] Task has been finished, but no results found in bytes [task.id=0114960b-2d97-49af-ba37-0965fb2062ea, task.hash=dcf0c7a894d7dbad8a329b51df4a50d9, organisation_id=mispoes, scheduler_id=boefje-mispoes]
[2023-07-12 14:54:32 +0000] [WARNING] [scheduler.schedulers.scheduler] Could not check if task is running: id='723d6f111a964ffebfb9ef73ed81c95f' boefje=Boefje(id='ssl-version', version=None) input_ooi='Website|internet|134.209.85.72|tcp|80|http|internet|www.mispo.es' organization='mispoes' dispatches=[] [organisation_id=mispoes, scheduler_id=boefje-mispoes, caller=push_tasks_for_random_objects]
Traceback (most recent call last):
  File "/app/scheduler/scheduler/schedulers/boefje.py", line 483, in push_task
    is_running = self.is_task_running(task)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/scheduler/scheduler/schedulers/boefje.py", line 441, in is_task_running
    raise RuntimeError("Task has been finished, but no results found in bytes")
RuntimeError: Task has been finished, but no results found in bytes
[2023-07-12 14:54:32 +0000] [ERROR] [scheduler.schedulers.scheduler] Task has been finished, but no results found in bytes [task.id=9fa430da-8554-46f0-a8b1-505a1dc7df4f, task.hash=97d6f0a1603e815c0cd6c71802ceabf7, organisation_id=mispoes, scheduler_id=boefje-mispoes]
[2023-07-12 14:54:32 +0000] [WARNING] [scheduler.schedulers.scheduler] Could not check if task is running: id='e06faf2f271d483ba453a18b448898e8' boefje=Boefje(id='ssl-certificates', version=None) input_ooi='Website|internet|134.209.85.72|tcp|80|http|internet|www.mispo.es' organization='mispoes' dispatches=[] [organisation_id=mispoes, scheduler_id=boefje-mispoes, caller=push_tasks_for_random_objects]
Traceback (most recent call last):
  File "/app/scheduler/scheduler/schedulers/boefje.py", line 483, in push_task
    is_running = self.is_task_running(task)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/scheduler/scheduler/schedulers/boefje.py", line 441, in is_task_running
    raise RuntimeError("Task has been finished, but no results found in bytes")
RuntimeError: Task has been finished, but no results found in bytes
```
These errors are not actionable and I have no idea what's wrong.

Because a stack trace is logged, this doesn't seem to be expected behaviour for a normal OpenKAT installation.

**Comment by jpbruinsslot** [comment]
This is likely the result of a difference between the state of the scheduler and bytes. In this case the scheduler gets the notion of a task that has been finished but there is no bytes data to be found. Was the state of bytes cleared perhaps?

It could also be that the references might be incorrect somehow, I'll have to look into that.

**Comment by praseodym** [comment]
> Was the state of bytes cleared perhaps?

No, we never removed any data from this environment.

**Comment by jpbruinsslot** [comment]
Ok, I'll investigate

**Comment by jpbruinsslot** [comment]
> > Was the state of bytes cleared perhaps?
> 
> No, we never removed any data from this environment.

Are there perhaps any helpful bytes logging available?

**Comment by jpbruinsslot** [comment]
I've looked at the logs. The scheduler mentions that the task in the scheduler itself is finished. Meaning the scheduler received a signal to set the task to completed or failed. It will check before it will rerun a task if that task is still being worked on. It will look at the internal state representation in the database and it will look at bytes to determine that it has been finished. When bytes returns that it isn't able to find the task. Then there is something wrong.

I couldn't find something in the logging in bytes that might tell the story why there isn't anything available in bytes. That likely happened before the restart of bytes.

At the moment the scheduler will disregard the task and not schedule it before the bytes issue has been resolved. What I can do though is to check if the task has been disregarded for a day, and reschedule it regardless when it is a rescheduling request.
